shared/pubsub.js

```diff
@@ -91,7 +92,15 @@ export const REQUEST_TYPE = Object.freeze({
 
 export const RESPONSE_TYPE = Object.freeze({
   ERROR: 'error',
-  SUCCESS: 'success'
+  SUCCESS: 'success',
+  PUSH_ERROR: 'push-error'
+})
+
+export const PUSH_SERVER_ACTION_TYPE = Object.freeze({
+  SEND_PUBLIC_KEY: 'send-public-key',
+  STORE_SUBSCRIPTION: 'store-subscription',
+  DELETE_SUBSCRIPTION: 'delete-subscription',
+  SEND_PUSH_NOTIFICATION: 'send-push-notification'
```

Comment:
Right, but that would require `server.js` to have the corresponding message handlers for all of `REQUEST_TYPE.PUSH_` keys. Instead, I think it's cleaner for it to have one message handler for `REQUEST_TYPE.PUSH_ACTION` and then this handler takes the appropriate action depending on which action it is (passed as `actionType` field in the payload)

Besides that, why not using these keys as `PUSH_ACTION.SOMETHING` instead of `PUSH_SOMETHING` if we want to make it clear they are push-notification related? :)

Btw,

> Since these message types are used to make the server do something, then maybe they could have been added as keys into the REQUEST type

This made me realise that it makes more sense to put `PUSH_ACTION` under `REQUEST_TYPE` instead of `NOTIFICATION_TYPE` which is where it currently is, so move it there.